addon-test-support/RouteInfoMock.ts

```diff
@@ -0,0 +1,75 @@
+interface RouteInfo {
+  readonly name: string;
+  readonly parent: RouteInfo | null;
+  readonly child: RouteInfo | null;
+  readonly localName: string;
+  readonly params: { [name: string]: unknown } | undefined;
+  readonly paramNames: string[];
+  readonly queryParams: { [name: string]: string };
+  readonly metadata: unknown;
+  find(
+    predicate: (this: any, routeInfo: RouteInfo, i: number) => boolean,
+    thisArg?: any
+  ): RouteInfo | undefined;
+}
+
+export default class RouteInfoMock implements RouteInfo {
+  name: string;
+  parent: RouteInfo | null = null;
+  child: RouteInfo | null = null;
+  localName: string;
+  params: { [name: string]: unknown } = {};
+  paramNames: string[] = [];
+  queryParams: { [name: string]: string } = {};
+  metadata: unknown;
+
+  constructor(name: string) {
+    this.name = name;
+    this.localName = name.split('.').pop()!;
+  }
+
+  withParent(parent: RouteInfo): RouteInfoMock {
+    this.parent = parent;
+    return this;
+  }
+
+  withChild(child: RouteInfo): RouteInfoMock {
+    this.child = child;
+    return this;
+  }
+
+  withParams(params: { [name: string]: unknown }): RouteInfoMock {
+    this.params = params;
+    this.paramNames = Object.keys(params);
+
+    return this;
+  }
+
+  withQueryParams(queryParams: { [name: string]: string }): RouteInfoMock {
+    this.queryParams = queryParams;
+    return this;
+  }
+
+  withMetadata(metadata: unknown): RouteInfoMock {
+    this.metadata = metadata;
+    return this;
+  }
+
+  find(
+    predicate: (this: any, routeInfo: RouteInfo, i: number) => boolean,
+    thisArg?: any
```

Comment:
A couple notes:

- We want to *not* have `any` here, and the `this` must always be a `RouteInfo`.
- We should need to actually match the original type for the `predicate`, so:

```suggestion
    predicate: (this: RouteInfo, routeInfo: RouteInfo, idx: number, array?: RouteInfo[]) => boolean,
    thisArg?: RouteInfo
```

(I know [the actual implementation](https://github.com/tildeio/router.js/blob/4f5d411e9ba44efea003bc3159484b4060ebfc2a/lib/router/route-info.ts#L79-L99) has `any`s there, but 😭.)